### PR TITLE
#21 Logging websocket

### PR DESCRIPTION
**Pull request by jake-lewis**

### CODE REVIEW

**Comment by jake-lewis** [comment]
This is only covering websocket registration, use of websockets will have to be reworked after this has been merged

File: src/main/java/com/logdyn/api/endpoints/LoggingEndpoint.java [hunk 1]
```diff
@@ -46,22 +48,27 @@
 	
 	private static final Level DEFAULT_LEVEL = Level.FINE;
 	
-	private Session session;
+	private Session websocketSession;
+	private UUID websocketUUID;
```
Comment (Mattihew) [inline src/main/java/com/logdyn/api/endpoints/LoggingEndpoint.java]: websocketUUID does not need to be a field

File: src/main/java/com/logdyn/api/endpoints/LoggingEndpoint.java [hunk 2]
```diff
@@ -117,19 +124,58 @@ public void onMessage(final Reader jsonString)
...
+		{
+			username = userPrincipal.getName();
+		}
+		
+		if (null != username)
+		{
+			set = LoggingEndpoint.ENDPOINT_USAGE.get(username);			
+			if (null == set)
+			{
+				set = new ConcurrentHashMap<Session, Void>().keySet();
```
Comment (Mattihew) [inline src/main/java/com/logdyn/api/endpoints/LoggingEndpoint.java]: should instead be `Collections.newSetFromMap(new ConcurrentHashMap<>()`

File: src/main/java/com/logdyn/api/endpoints/LoggingEndpoint.java [hunk 3]
```diff
@@ -117,19 +124,58 @@ public void onMessage(final Reader jsonString)
...
 	}
+	
+	/**
+	 * Used when initially opening a websocket, adds websocket to list of endpoints
+	 * @param websocketId
+	 * @param httpSessionId
+	 */
+	public static void registerWebsocket(final UUID websocketId, final String httpSessionId)
+	{
+		final Session websocketSession = LoggingEndpoint.ENDPOINTS.get(websocketId);
```
Comment (Mattihew) [inline src/main/java/com/logdyn/api/endpoints/LoggingEndpoint.java]: change `get` to `remove`

File: src/main/java/com/logdyn/api/endpoints/LoggingEndpoint.java [hunk 4]
```diff
@@ -117,19 +124,58 @@ public void onMessage(final Reader jsonString)
...
+				set = new ConcurrentHashMap<Session, Void>().keySet();
+				LoggingEndpoint.ENDPOINT_USAGE.put(username, set);
+			}			
+			set.add(websocketSession);
+		}
+		
+		set = LoggingEndpoint.ENDPOINT_USAGE.get(httpSessionId);		
+		if (null == set)
+		{
+			set = new ConcurrentHashMap<Session, Void>().keySet();
```
Comment (Mattihew) [inline src/main/java/com/logdyn/api/endpoints/LoggingEndpoint.java]: should instead be `Collections.newSetFromMap(new ConcurrentHashMap<>()`

**Review by Mattihew — CHANGES_REQUESTED** [review]

**Review by Mattihew — APPROVED** [review]

**Pull request merged**